Rpackage/R/fitAndScore.R

```diff
@@ -0,0 +1,191 @@
+fit_and_score <- structure(function(tree, target.mat, feature.mat, 
+                                    parameters, feature_names=NULL, 
+                                    n_folds = 3, scorer=NULL, loss = "hinge",
+                                    pruning = TRUE){
+  ### total length
+  l <- length(target.mat[, 1])
+  
+  ### train and test index for each fold
+  fold_split_idx <- NULL
+  
+  ### initial row
+  fold_split_idx$train <- seq(from = 1, to = ((l/n_folds)*(n_folds-1)), by = 1)
+  fold_split_idx$test <- seq(from = ((l/n_folds)*(n_folds-1)) + 1, to = l, by = 1)
+  
+  ### middle values
+  for(n in (n_folds-1):2){
+    this_row_train <- c(seq(from = 1, to = ((l/n_folds)*(n-1)), by = 1), seq(from = as.integer((l/n_folds)*(n)+1), to = l, by = 1))
+    this_row_test <- seq(from = ((l/n_folds)*(n-1)) + 1, to = as.integer((l/n_folds)*(n)))
+    
+    fold_split_idx$train <- rbind(fold_split_idx$train, this_row_train)
+    fold_split_idx$test <- rbind(fold_split_idx$test, this_row_test)
+  }
+  
+  ### end row
+  this_row_train <- seq(from = (as.integer((l/n_folds)+1)), to = l, by = 1)
+  this_row_test <- seq(from = 1, to = as.integer((l/n_folds)), by = 1)
+  
+  fold_split_idx$train <- rbind(fold_split_idx$train, this_row_train)
+  fold_split_idx$test <- rbind(fold_split_idx$test, this_row_test)
+  
+  ### fold trees
+  fold_tree <- NULL
+  for(i in 1:n_folds){
+    fold_tree[[i]] <- mmit(target.mat[fold_split_idx$train[i,],], feature.mat[fold_split_idx$train[i,],],  
+                                       maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                                       loss = loss, min_sample = as.numeric(parameters$min_sample))
+  }
+  
+  ### master tree
+  master_tree <- mmit(target.mat, feature.mat,  
+                      maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                      loss = loss, min_sample = as.numeric(parameters$min_sample))
+  
+  
+  ### if pruning
+  if(pruning){
+    # Get the pruned master and cross-validation trees
+    master_data <- pruning(master_tree)
+    master_alphas <- master_data[, 1]
+    master_pruned_trees <- master_data[, -1]
+    
+    fold_alphas <- NULL
+    fold_prune_trees <- NULL
+    for(i in 1 : n_folds){
+      fold_data <- pruning(fold_tree[[i]])
+      fold_alphas[[i]] <- fold_data[, 1] 
+      fold_prune_trees[[i]] <- fold_data[, -1]
+    }
+    
+    ### alphas list should not contain repeating alpha
```

Comment:
If this happens, it's because there is a bug in your code. In the present form, the code would silently continue, which is incorrect.

Reply:
done